### PR TITLE
bpo-44439: BZ2File.write() / LZMAFile.write() handle buffer protocol correctly

### PR DESCRIPTION
**Pull request by ghost**
The underlying C codes handle _buffer protocol objects_ correctly, so only fix the wrong length in Python code is fine.

https://bugs.python.org/issue44439

### CODE REVIEW

File: Lib/test/test_gzip.py [hunk 1]
```diff
@@ -592,6 +592,14 @@ def test_prepend_error(self):
         with gzip.open(self.filename, "rb") as f:
             f._buffer.raw._fp.prepend()
 
+    def test_issue44439(self):
+        q = array.array('Q', [1, 2, 3, 4, 5])
+
+        with gzip.GzipFile(fileobj=io.BytesIO(), mode='w') as f:
+            f.write(q)
```
Comment (serhiy-storchaka) [inline Lib/test/test_gzip.py]: Test the result of write.
```
self.assertEqual(f.write(q), ...
```

File: Lib/bz2.py [hunk 2]
```diff
@@ -218,15 +218,22 @@ def readlines(self, size=-1):
     def write(self, data):
         """Write a byte string to the file.
 
-        Returns the number of uncompressed bytes written, which is
-        always len(data). Note that due to buffering, the file on disk
-        may not reflect the data written until close() is called.
+        Returns the number of uncompressed bytes written. Note that
```
Comment (serhiy-storchaka) [inline Lib/bz2.py]: One detail is missed, present in the old docstring. That all bytes are written.

In the docstring of `BufferedWriter.write`:
```
      Returns the number of bytes written, which is always the length of b in bytes.
```
I think this wording can be borrowed.

File: Lib/test/test_gzip.py (hunk 1 above)
Comment (ghost) [reply Lib/test/test_gzip.py]: Added.

File: Lib/bz2.py (hunk 2 above)
Comment (ghost) [reply Lib/bz2.py]: Added.

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Comment by animalize** [comment]
`3.9` branch has a code conflict, I created PR https://github.com/python/cpython/pull/26846 for `3.9` branch.

**Pull request merged**